### ISSUE TITLE
Juju secrets documentation lacks detail and is misleading in some places

### ISSUE THREAD

**Issue by DnPlas** [juju/juju#18892]
The docs around juju secrets are lacking some details and information that is relevant for using them, specially when integrating with charms. Here are some of the things I have run into while trying to integrate juju secrets with a charm:

1. How to integrate user secrets with charms - While there are some WARNING messages in [How to manage secrets](https://juju.is/docs/juju/manage-secrets) telling users that "For the application to be able to use the secret, it needs to be configured with the secret URI. " is not enough to set a clear view of how to actually integrate with a charm. In https://github.com/canonical/istio-operators/issues/380#issuecomment-2020861148 @jameinel gave an example of how to use user secrets in a charm, specifying details about the things a charm MUST have before a secret can be used. I think this  information is not obvious and could be of great help to have it in the docs.
2. Secret events from user secrets - I don't think we have any documentation explaining how a charm that wants to use a user secret can subscribe to Secret Events. It is sort of related to the implementation details sated above, but I think a nice explanation that says what has to happen before a charm can subscribe to those events is really important, specially since this events are not triggered from commands like `juju grant-secret <secret-name> <application-name>`, contrasting to the behaviour of `juju relate` or `juju config`.
3. Secret events - The [Secret events trigger](https://juju.is/docs/sdk/secret-events#heading--secret-event-triggers) section in the Secret events docs may be confusing if no details are given about these events triggers. I think a good addition to this information would be to describe how events are triggered, i.e. what has to happen for the charm to be able to observe user or charm owned secrets.
4. Add secrets to a charm - For someone who is not familiar with juju secrets, finding out that this guide [How to add secrets to a charm](https://juju.is/docs/sdk/add-a-secret-to-a-charm) is the only one that explains how to add secrets to a charm can lead to a lot of confusion, as it is limited to one use case for using secrets. I think it would be nice to have different guides for different use cases as stated previously and perhaps change the title to avoid thinking this is the only way to add secrets to a charm.

**Comment by tmihoc** [comment]
@wallyworld assigning this to you as this is already something you've been working on in https://canonical-juju.readthedocs-hosted.com/en/latest/user/reference/hook/#secret-hooks and https://canonical-juju.readthedocs-hosted.com/en/latest/user/reference/hook/#secret-changed and the sections after it. The todo at this point is to check that we now address all the four points raised above. 

@tonyandrewmeyer should be able to help with the Ops end of things as he's worked on https://ops.readthedocs.io/en/latest/howto/manage-secrets.html

**Comment by tmihoc** [comment]
@tonyandrewmeyer for example it feels like the note at the top of this section https://ops.readthedocs.io/en/latest/howto/manage-secrets.html#secret-observer-charm could use a link to Manage configurations, and our Manage configurations doc could use an example showing a secret, maybe in a dropdown.

**Comment by tmihoc** [comment]
Our documentation has undergone significant improvements since this issue was open, on both the Juju and the Charmcraft and Ops end:

https://documentation.ubuntu.com/juju/3.6/reference/secret/
https://documentation.ubuntu.com/juju/3.6/reference/hook/#secret-hooks
https://documentation.ubuntu.com/charmcraft/stable/howto/manage-charms/#manage-secrets
https://documentation.ubuntu.com/ops/latest/howto/manage-secrets/ 

Still, the Juju concept doc only covers lifecycle for a charm secret: https://documentation.ubuntu.com/juju/3.6/reference/secret/#secret-lifecycle
The Charmcraft bit fails to clarify the config path is just for user secrets: https://documentation.ubuntu.com/charmcraft/stable/howto/manage-charms/#manage-secrets
The Ops how-to only covers charm secrets: https://documentation.ubuntu.com/ops/latest/howto/manage-secrets/ 

Overall, we're still missing some signposting, so keeping this issue open till we've connected all the dots.

**Comment by tmihoc** [comment]
Ian says: About secret lifecycle for user secrets: The only hook that's relevant is the secret-changed hook, which observers will get if the user changes the secret.

About the Charmcraft and Ops gaps: I'll just create issues in those projects.

**Comment by tonyandrewmeyer** [comment]
I opened doc PRs to hopefully close out the remaining items:

 - **canonical/operator** — [PR link](https://github.com/canonical/operator/pull/2522): Adds a dedicated "User-secret observer charm" section to `docs/howto/manage-secrets.md`, documenting the prerequisites, code pattern, and the `secret-changed`-only event contract (points 1, 2, 3).
-  **juju/juju** — [PR link](https://github.com/juju/juju/pull/22529): Extends the `## Secret lifecycle` section in `docs/reference/secret.md` to cover the user-secret lifecycle alongside the charm-secret lifecycle, and adds a clarifying note in `docs/reference/hook.md` under `secret-changed` (points 2, 3).
 - **canonical/charmcraft** — [PR link](https://github.com/canonical/charmcraft/pull/2712): Adds a use-case map to the Manage secrets section of `docs/howto/manage-charms.rst`, making clear that the `type: secret` config-option path is specifically for user secrets (point 4).

The key fact stated consistently across all three: for a user secret the only hook is `secret-changed` (fired when the user updates the secret); `juju grant-secret` does not fire a hook.

**Comment by tmihoc** [comment]
Thank you very much, @tonyandrewmeyer !

**Comment by tmihoc** [comment]
Fixed in the last round of PRs.
